cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -766,10 +774,10 @@ private SearchResults searchByCustomIndex(SearchRequest request) throws BadReque
         while ((next = scanner.next()) != null && results.size() < fetchSize) {
           Optional<MetadataEntry> metadataEntry =
             parseRow(next, column, request.getTypes(), request.shouldShowHidden());
-          if (!metadataEntry.isPresent()) {
+          if (metadataEntry == null) {
```

Comment:
Why check for `null` instead? Usually when `Optional` is used, it shouldn't be `null`.

Reply:
Good point, switched back in [#11489](https://github.com/cdapio/cdap/pull/11489)